### PR TITLE
dash/datagrid-edit-events

### PR DESCRIPTION
**Pull request by sebastianbochan**
Added API event to column sorting and cell editing.

----
- [x] - test
- [x] - cell editing
- [x] - column sorting
- [x] - header events

### CODE REVIEW

File: ts/DataGrid/Actions/CellEditing.ts [hunk 1]
```diff
@@ -111,6 +116,9 @@ class CellEditing {
         }
 
         cell.setValue(submit ? newValue : cell.value, submit);
+
+        dataGrid.options?.events?.cell.afterEdit?.call(cell);
```
Comment (ddragula) [inline ts/DataGrid/Actions/CellEditing.ts:120]:
```suggestion
        dataGrid.options?.events?.cell?.afterEdit?.call(cell);
```

File: samples/data-grid/cypress/column-cell-events/demo.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            sorting: true
+        }
+    },
+    columns: {
+        product: {
+            editable: true,
+            events: {
+                afterSorting: function () {
+                    doc.getElementById('columnSorting').value = 'afterSorting';
+                }
```
Comment (ddragula) [inline samples/data-grid/cypress/column-cell-events/demo.js:24]: I would move it to the global events options. @stitot maybe you have any suggestions?

**Review by ddragula — CHANGES_REQUESTED** [review]
1. There is already a demo of `cell-events`, maybe they could be unified?
2. Shouldn't we additionally emit events on the `dataGrid` object?

**Comment by sebastianbochan** [comment]
1. It is, however, I would like to isolate cypress tests and basic examples. It's safer.

File: samples/data-grid/cypress/column-cell-events/demo.js (hunk 2 above)
Comment (sebastianbochan) [reply samples/data-grid/cypress/column-cell-events/demo.js:24]: IMO, we should not mix cell and column events. It's the same situation like series and point.

Comment (sebastianbochan) [reply samples/data-grid/cypress/column-cell-events/demo.js:24]: Do you mean to move `column.events` to `events.column`? Hm...

Comment (sebastianbochan) [reply samples/data-grid/cypress/column-cell-events/demo.js:24]: It sounds more reasonable and convinced me.

Let's change it.

File: ts/DataGrid/DataGridOptions.d.ts [hunk 3]
```diff
@@ -205,6 +211,11 @@ export interface ColumnOptions {
      * @default true
      */
     sorting?: boolean;
+
+    /**
+     * Events related to column.
```
Comment (karolkolodziej) [inline ts/DataGrid/DataGridOptions.d.ts:216]: Is it still true? Interface is called `DataGridCellEvents`

File: ts/DataGrid/DataGridOptions.d.ts [hunk 4]
```diff
@@ -259,6 +270,17 @@ export interface DataGridCellEvents {
...
+    /**
+     * Callback function to be called after editing of cell value.
+     */
+    afterEdit?: DataGridCellEventCallback;
+
+    /**
+     * Callback function to be called when the column is sorted
+     * (for instance, after clicking on header).
+     */
+    afterSorting?: DataGridColumnEventCallback;
```
Comment (karolkolodziej) [inline ts/DataGrid/DataGridOptions.d.ts:283]: I believe this should be called `DataGridCellEventCallback` right?

**Review by karolkolodziej — APPROVED** [review]

File: ts/DataGrid/DataGridOptions.d.ts (hunk 4 above)
Comment (sebastianbochan) [reply ts/DataGrid/DataGridOptions.d.ts:283]: Nope, because "this"  refers to column, not cell.

**Review by ddragula — APPROVED** [review]

**Pull request merged**